cpp/include/cudf/unary.hpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cudf.h"
+
+namespace cudf
+{
+
+/**
+ * @brief Types of unary operations that can be performed on data.
+ */
+enum unary_op{
+  SIN,          ///< Trigonometric sine
+  COS,          ///< Trigonometric cosine
+  TAN,          ///< Trigonometric tangent
+  ARCSIN,       ///< Trigonometric sine inverse
+  ARCCOS,       ///< Trigonometric cosine inverse
+  ARCTAN,       ///< Trigonometric tangent inverse
+  EXP,          ///< Exponential (base e, Euler number)
+  LOG,          ///< Natural Logarithm (base e)
+  SQRT,         ///< Square-root (x^0.5)
+  CEIL,         ///< Smallest integer value not less than arg
+  FLOOR,        ///< largest integer value not greater than arg
+  ABS,          ///< Absolute value
+  BIT_INVERT,   ///< Bitwise Not (~)
+  NOT,          ///< Logical Not (!)
+  INVALID_UNARY ///< invalid operation
```

Comment:
What's the purpose of having an `INVALID_UNARY` option? Is this needed?

Reply:
Blazing added this for whatever reason